jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/MultiReleaseJarFile.java

```diff
@@ -14,109 +14,59 @@
 package org.eclipse.jetty.util;
 
 import java.io.Closeable;
-import java.io.File;
 import java.io.IOException;
-import java.io.InputStream;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.Locale;
-import java.util.Map;
-import java.util.TreeMap;
-import java.util.jar.JarEntry;
-import java.util.jar.JarFile;
-import java.util.jar.Manifest;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
 import java.util.stream.Stream;
 
+import org.eclipse.jetty.util.resource.Resource;
+
 /**
  * <p>Utility class to handle a Multi Release Jar file</p>
  */
 public class MultiReleaseJarFile implements Closeable
 {
-    private static final String META_INF_VERSIONS = "META-INF/versions/";
-
-    private final JarFile jarFile;
-    private final int platform;
-    private final boolean multiRelease;
-
-    /* Map to hold unversioned name to VersionedJarEntry */
-    private final Map<String, VersionedJarEntry> entries;
+    private final Path jarFile;
+    private final Resource.Mount jarResource;
 
     /**
      * Construct a multi release jar file for the current JVM version, ignoring directories.
      *
-     * @param file The file to open
+     * @param jarFile The file to open
      * @throws IOException if the jar file cannot be read
      */
-    public MultiReleaseJarFile(File file) throws IOException
+    public MultiReleaseJarFile(Path jarFile) throws IOException
     {
-        this(file, JavaVersion.VERSION.getPlatform(), false);
-    }
+        Objects.requireNonNull(jarFile, "Jar File");
 
-    /**
-     * Construct a multi release jar file
-     *
-     * @param file The file to open
-     * @param javaPlatform The  JVM platform to apply when selecting a version.
-     * @param includeDirectories true if any directory entries should not be ignored
-     * @throws IOException if the jar file cannot be read
-     */
-    public MultiReleaseJarFile(File file, int javaPlatform, boolean includeDirectories) throws IOException
```

Comment:
Why is the constructor accepting `javaPlatform` gone?

Reply:
It wasn't used.